From: Robert Schwebel <r.schwebel@pengutronix.de>
Date: Sat, 29 Oct 2011 19:14:59 +0200
Subject: [PATCH] make eggdbus crosscompile aware

The build system compiles tools like eggdbus-glib-genmarshal and
eggdbus-binding-tool; if you cross compile the package, these fines are
built for the "host" architecture, not for the "build" one, but the
Makefile tries to run the cross compiled binaries then.

Make it possible to overwrite them from the configure command line.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>
---
 configure.ac            |   10 ++++++++++
 src/eggdbus/Makefile.am |    2 +-
 2 files changed, 11 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 497cb7f..a4e792a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -34,6 +34,16 @@ AM_CONDITIONAL(MAN_PAGES_ENABLED, test x$enable_man_pages = xyes)
 
 GTK_DOC_CHECK([1.3])
 
+EGGDBUS_TOOLS=""
+AC_ARG_WITH(eggdbus-tools,
+  AS_HELP_STRING([--with-eggdbus-tools],[path to eggdbus-binding-tool (for cross compilation)]),
+[
+  EGGDBUS_TOOLS="$withval"
+],[
+  EGGDBUS_TOOLS="\$(top_builddir)/src/eggdbus"
+])
+AC_SUBST(EGGDBUS_TOOLS)
+
 #### gcc warning flags
 
 if test "x$GCC" = "xyes"; then
diff --git a/src/eggdbus/Makefile.am b/src/eggdbus/Makefile.am
index aebb984..95615cd 100644
--- a/src/eggdbus/Makefile.am
+++ b/src/eggdbus/Makefile.am
@@ -130,7 +130,7 @@ BUILT_SOURCES =								\
 	$(NULL)
 
 eggdbus-built-sources.stamp : eggdbus-binding-tool Makefile.am org.freedesktop.DBus.xml org.freedesktop.DBus.Peer.xml org.freedesktop.DBus.Properties.xml org.freedesktop.DBus.Introspectable.xml eggdbus-glib-genmarshal
-	EGG_DBUS_GENMARSHAL=$(top_builddir)/src/eggdbus/eggdbus-glib-genmarshal $(top_builddir)/src/eggdbus/eggdbus-binding-tool							\
+	EGG_DBUS_GENMARSHAL=$(EGGDBUS_TOOLS)/eggdbus-glib-genmarshal $(EGGDBUS_TOOLS)/eggdbus-binding-tool							\
 		--namespace EggDBus									\
 		--dbus-namespace org.freedesktop.DBus							\
 		--interfaces-only									\
